images/cluster-setup/install.sh

```diff
@@ -20,122 +20,127 @@ set -o pipefail
 
 usage() {
 
-    printf "Usage: DATA_DIR="/workspace" ./register.sh \n\n"
-
-    # Parameters
-    printf "The following parameters need to be passed to the script: \n"
-    printf "DATA_DIR: the location of the cluster files \n"
+  printf "
+Usage:
+    %s [options]
+
+Deploy Pipelines Service on the clusters as per the configuration in
+WORKSPACE_DIR.
+
+Optional arguments:
+    -w, --workspace-dir WORKSPACE_DIR
+        Location of the folder holding the clusters configuration.
+        Default: \$WORKSPACE_DIR or current directory if the environment
+        variable is unset.
+    -d, --debug
+        Activate tracing/debug mode.
+    -h, --help
+        Display this message.
+
+Example:
+    %s --workspace_dir WORKSPACE_DIR
+" "${0##*/}" "${0##*/}" >&2
 }
 
-prechecks () {
-    DATA_DIR=${DATA_DIR:-}
-    if [[ -z "${DATA_DIR}" ]]; then
-        printf "DATA_DIR not set\n\n"
-        usage
-	      exit 1
-    fi
+parse_args() {
+  WORKSPACE_DIR="${WORKSPACE_DIR:-$PWD}"
+
+  while [[ $# -gt 0 ]]; do
+    case $1 in
+    -w | --workspace-dir)
+      shift
+      WORKSPACE_DIR="$1"
+      ;;
+    -d | --debug)
+      set -x
+      ;;
+    -h | --help)
+      usage
+      exit 0
+      ;;
+    *)
+      echo "Unknown argument: $1"
+      usage
+      exit 1
+      ;;
+    esac
+    shift
+  done
 }
 
-# populate clusters with the cluster names taken from the kubeconfig
-# populate contexts with the context name taken from the kubeconfig
-# populate kubeconfigs with the associated kubeconfig for each cluster name
-# only consider the first context for a specific cluster
-get_clusters() {
-    clusters=()
-    contexts=()
-    kubeconfigs=()
-    printf "Extracting files under the kubeconfig dir and reading the content in each file \n"
-    files=("$(ls "$DATA_DIR/gitops/sre/credentials/kubeconfig/compute")")
-    for kubeconfig in "${files[@]}"; do
-        printf "%s\n" "$kubeconfig"
-        subs=("$(KUBECONFIG=${DATA_DIR}/gitops/sre/credentials/kubeconfig/compute/${kubeconfig} kubectl config view -o jsonpath='{range .contexts[*]}{.name}{","}{.context.cluster}{"\n"}{end}')")
-        for sub in "${subs[@]}"; do
-            context=$(echo -n "${sub}" | cut -d ',' -f 1)
-            cluster=$(echo -n "${sub}" | cut -d ',' -f 2 | cut -d ':' -f 1)
-	    if ! (echo "${clusters[@]}" | grep "${cluster}"); then
-                clusters+=( "${cluster}" )
-                contexts+=( "${context}" )
-                kubeconfigs+=( "${kubeconfig}" )
-                printf "%s --- %s --- %s \n" "$cluster" "$context" "$kubeconfig"
-            fi
-        done
-    done
+init() {
+  cluster_config_dirs=($(
+    find "${WORKSPACE_DIR}/environment/compute" \
+      -name kustomization.yaml \
+      -exec dirname {} \;
+  ))
+}
```

Comment:
As explained in the other PR the existing approach is more flexible.

Reply:
I thought that we had an understanding, but it looks like I was wrong. Let's have another talk about it to make sure I understand where we stand on this topic.

Reply:
Reverted to old behavior.